src/pybind/mgr/cephadm/module.py

```diff
@@ -472,9 +469,8 @@ def notify(self, notify_type, notify_id):
         if notify_type == "mon_map":
             # get monmap mtime so we can refresh configs when mons change
             monmap = self.get('mon_map')
-            self.last_monmap = datetime.datetime.strptime(
-                monmap['modified'], CEPH_DATEFMT)
-            if self.last_monmap and self.last_monmap > datetime.datetime.utcnow():
+            self.last_monmap = str_to_datetime(monmap['modified'])
```

Comment:
I mean, if it works, that's great!

Reply:
The function `str_to_datetime` can parse multiple formats that are used in Ceph now and in the past.